src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs

```diff
@@ -302,20 +303,19 @@ public async Task<IOperationResult> Execute(Action<IOperationProgress> onProgres
                     details: new ExceptionDetails(e));
                 _serverStore.NotificationCenter.Add(alert);
 
-                var databaseRecord = _serverStore.LoadDatabaseRecord(RestoreFromConfiguration.DatabaseName, out _);
-
-                if (databaseRecord == null)
-                {
-                    // delete any files that we already created during the restore
-                    IOExtensions.DeleteDirectory(RestoreFromConfiguration.DataDirectory);
-                }
-                else
+                using (_serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+                using (context.OpenReadTransaction())
                 {
-                    databaseRecord.Disabled = false;
-                    databaseRecord.DatabaseState = DatabaseStateStatus.Normal;
-
-                    var deleteResult = await _serverStore.DeleteDatabaseAsync(RestoreFromConfiguration.DatabaseName, true, new[] { _serverStore.NodeTag }, RaftIdGenerator.DontCareId);
-                    await _serverStore.Cluster.WaitForIndexNotification(deleteResult.Index);
+                    if (_serverStore.Cluster.DatabaseExists(context, RestoreFromConfiguration.DatabaseName) == false)
+                    {
+                        // delete any files that we already created during the restore
+                        IOExtensions.DeleteDirectory(RestoreFromConfiguration.DataDirectory);
+                    }
+                    else
+                    {
+                        var deleteResult = await _serverStore.DeleteDatabaseAsync(RestoreFromConfiguration.DatabaseName, true, new[] { _serverStore.NodeTag }, RaftIdGenerator.DontCareId);
+                        await _serverStore.Cluster.WaitForIndexNotification(deleteResult.Index);
```

Comment:
Holding the read tx for a long time is a bad idea, since we will keep the journals/scartch/temp files of it and any write tx will allocate more and files until this tx will be disposed.

Reply:
d o n e